### DISCUSSION TITLE
Create 3 solid red light and not connecting to wifi

### DISCUSSION THREAD

**Discussion by gregdodd** [iRobotEducation/create3_docs#183] [Questions] [answered]
My create 3 robot is giving me a solid red light on start up. It stays red for about 2 mins and then restarts with a green/yellow light. It will not connect to wifi. It never has the solid white light only either Red or the Green/Yellow.

I've tried several hard restarts (removing battery for 15+mins and reinstalling), but get the same result.
Logs attached.
[messages.txt](https://github.com/iRobotEducation/create3_docs/files/9454815/messages.txt)

**Comment by maddiepero** [comment]
Hi, we have had similar, but slightly less intense, connectivity issues with the robots we have Tufts. I have noticed it may take two or three tries to connect the robot to wifi so definitely keep returning to the setup page and attempting to connect it. However, someone on our team with better knowledge of the logs will hopefully take a look at those since we have never experienced a red light right from the start-up. Has it been acting this way since you took it out of the box or have you been using it for a while?

**Comment by gregdodd** [comment]
When initially set it up it worked fine. The idle light was the expected solid white light. I did some of the tutorial activities and then put it in deep sleep mode (holding power button for 10 seconds) and when I woke it up about 3 weeks later it immediately started acting up with the constant yellow/green light, not connecting etc.

**Reply by maddiepero** [reply]
I'm glad to hear it worked fine out of the box! You don't happen to have multiple devices that are running ROS on your wifi network? We have had this problem most often when we have many robots on the same network. The logs also say "script called in periodic timing mode" this caught my eye because I have not seen this when we have connectivity issues.

**Reply by justinIRBT** [reply]
Are you running the webserver to connect to a wifi network when you are getting the yellow/green light?  Did you try connecting to a different wifi network than you succeeded at connecting to originally?  The robot should retain its wifi network through power down and restart.  Here are the lightring codes to confirm which errors you are getting:
https://iroboteducation.github.io/create3_docs/hw/face/

**Reply by gregdodd** [reply]
Yes I run the webserver and enter the wifi information, then it does the 'spinning cyan' attempting to connect light for ~10 seconds before the light suddenly turns off all together and it reboots. When it's done rebooting I have a solid yellow light. I've tried connecting to 3 different wifi networks with the same result.

**Reply by justinIRBT** [reply]
When you say "it reboots" what are you seeing that makes you think it is rebooting?  Are you connecting to wifi routers that the robot shows in the drop down list?  The wifi only supports 2.4GHz and the name can be case sensitive, so it is best to choose networks from the drop down list to ensure it can see it and the case is correct.  You can hit the re-scan button if the network is 2.4, but not in the list when you first load the page.  Also, are you running the latest G.2.2 release?

**Reply by gregdodd** [reply]
Only connecting to wifi routers from the drop down list. I'm running G.2.2 and tried reinstalling to see if that would fix it.

Here's the order of events when I attempt to connect to wifi (used the same process for 3 different routers): 
1. Select wifi network from dropdown and enter password. Receive successfully sent provision message.
2. Blue cyan light begins spinning indicating that it's trying to connect.
3. The robot gives a chime and the light goes off for a second (no green flash indicating a successful connection.
4. The spinning white boot up light begins spinning.
5. Robot chimes after bootup and solid yellow light shows. Once it gave a red light for ~1 second and then changed to yellow.

**Reply by justinIRBT** [reply]
Can you attach the log from the webserver.  I noticed in the original log you attached, your robot successfully got an IP address:
```
Aug 30 09:41:09 iRobot-88FABEC7371041B9890F5EA20874E018 daemon.err udhcpc[5330]: sending discover
Aug 30 09:41:09 iRobot-88FABEC7371041B9890F5EA20874E018 daemon.err udhcpc[5330]: sending select for 192.168.86.23
Aug 30 09:41:09 iRobot-88FABEC7371041B9890F5EA20874E018 daemon.err udhcpc[5330]: lease of 192.168.86.23 obtained, lease time 86400
```

**Reply by gregdodd** [reply]
Interesting! I can access the webserver through that IP so it is connected. Light is still solid yellow. Most recent logs are attached (I've tried connecting several times today as noted in the log activity).
[messages.txt](https://github.com/iRobotEducation/create3_docs/files/9464800/messages.txt)

**Reply by justinIRBT** [reply]
While I am looking at the log, can you try powering off the robot off the dock with the 10 second power button hold, then put back on the dock for a fresh boot and try accessing that IP address without putting the robot into softAP mode.  I am curious what the light shows after it finishes booting from a full power cycle while sitting stationary on the dock.

**Reply by gregdodd** [reply]
👍 will try that now.

**Reply by justinIRBT** [reply]
I am a little confused by some of the stuff I am seeing in the log.  Your robot seems to be failing its wheels test at startup, which generally happens if the wheels are moving when the test is run, but I assume your robot is idle on the floor.  I am seeing some odd behavior around the networking, I am curious if we can remove that from the equation just to make sure your robot is happy, then add networking back in.  The robot's internal network is also 192.168.x.x, but the third octet is different (192.168.186.2 and 192.168.10.1) for the internal address so I wouldn't expect an issue.  To wipe the networking, we can run the factory reset described:
https://iroboteducation.github.io/create3_docs/webserver/about/
There should be a hyperlink to factory reset.  Then try doing a power cycle reboot (hold power button off dock till robot powers down, then put on dock) and see if your robot comes up with a solid white light after playing the startup sound.  We can go from there.

**Reply by gregdodd** [reply]
Ok thanks, I'll try the factory reset and power cycle now. The light was still solid yellow after the 10 second power off and dock power on, but I could still connect to it through the ip address.

**Reply by gregdodd** [reply]
Another weird thing to note. When I pressed buttons 1 and 2 to put the robot in AP mode, I got a 1 second solid red light before the spinning cyan.

**Reply by gregdodd** [reply]
Post factory reset: light is still solid yellow, and I can't access the robot via the IP address anymore. Logs attached.
[messages.txt](https://github.com/iRobotEducation/create3_docs/files/9465041/messages.txt)

**Reply by justinIRBT** [reply]
Thanks, I am reviewing with the team and we will get back to you.  Just to confirm, the robot is idle on the dock on the floor when this startup issue is happening?

**Reply by gregdodd** [reply]
That’s correct

**Reply by gregdodd** [reply]
Thanks for your help!

**Reply by justinIRBT** [reply]
One more piece of information.  Can you provide the `memory` and `uptime` from the `About` page: https://iroboteducation.github.io/create3_docs/webserver/about/

**Reply by gregdodd** [reply]
Memory and uptime:
<img width="857" alt="Screen Shot 2022-08-31 at 6 00 40 PM" src="https://user-images.githubusercontent.com/31262364/187793096-e50c4eb1-baea-4b18-ac3c-bceedccde28c.png">

**Comment by justinIRBT** [comment, marked as answer]
Thanks for all the info.  I am not sure what is making your robot unhappy.  I am going to bump you up to the next level of support.

Please email:
[education@irobot.com](mailto:education@irobot.com)

In the email, please provide your order # and the robot serial number (`iRobot-88FABEC7371041B9890F5EA20874E018`) and reference this discussion thread as well as the fact that you were directed you to email them.  They should be able to take it from there.  Thanks for your patience!

**Reply by gregdodd** [reply]
Thanks for all your help Justin, I'll send them an email today.

**Reply by shamlian** [reply]
@gregdodd Checking in -- is everything working, now?

**Reply by gregdodd** [reply]
@shamlian thanks for checking in! I received a replacement unit a couple weeks ago and it is working as expected. I'll mark this discussion as answered. Apologies for the delay.

**Reply by shamlian** [reply]
No worries at all, glad to hear it. (And my apologies for your trouble!)
